cql3/statements/attach_service_level_statement.cc

```diff
@@ -43,6 +43,9 @@ attach_service_level_statement::execute(query_processor& qp,
         service::query_state &state,
         const query_options &,
         std::optional<service::group0_guard> guard) const {
+    if (guard) {
```

Comment:
> Currently guard is released immediately because those functions are
based on continuations and guard lifetime is not extended. In the following
commit we rewrite those functions to coroutines and lifetime will be
automatically extended. This would deadlock the client because we'd
try to take second guard inside auth code without releasing this unused
one.

This explicit calling out of special behavior is very helpful for reviewers. Thanks.